store-implementation/map-store/src/main/java/uk/gov/gchq/gaffer/mapstore/impl/DeleteElementsHandler.java

```diff
@@ -0,0 +1,167 @@
+import uk.gov.gchq.gaffer.operation.data.EntitySeed;
+import uk.gov.gchq.gaffer.operation.impl.delete.DeleteElements;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.ValidatedElements;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.SchemaElementDefinition;
+import uk.gov.gchq.gaffer.store.util.AggregatorUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An {@link OperationHandler} for the {@link DeleteElements} operation on the
+ * {@link MapStore}.
+ */
+public class DeleteElementsHandler implements OperationHandler<DeleteElements> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DeleteElementsHandler.class);
+
+  @Override
+  public Void doOperation(final DeleteElements deleteElements, final Context context, final Store store)
+      throws OperationException {
+    Iterable<? extends Element> elements = deleteElements.getInput();
+    if (deleteElements.isValidate()) {
+      elements = new ValidatedElements(elements, store.getSchema(), deleteElements.isSkipInvalidElements());
+    }
+
+    deleteElements(elements, (MapStore) store);
+    return null;
+  }
+
+  private void deleteElements(final Iterable<? extends Element> elements, final MapStore mapStore) {
+    final MapImpl mapImpl = mapStore.getMapImpl();
+    final Schema schema = mapStore.getSchema();
+
+    final int bufferSize = mapStore.getProperties().getIngestBufferSize();
+
+    if (bufferSize < 1) {
+      // Delete all elements directly
+      deleteBatch(mapImpl, schema, elements);
+    } else {
+      LOGGER.info("Deleting elements in batches, batch size = {}", bufferSize);
+      int count = 0;
+      final List<Element> batch = new ArrayList<>(bufferSize);
+      for (final Element element : elements) {
+        if (null != element) {
+          batch.add(mapImpl.cloneElement(element, schema));
+          count++;
+          if (count >= bufferSize) {
+            deleteBatch(mapImpl, schema, AggregatorUtil.ingestAggregate(batch, schema));
+            batch.clear();
+            count = 0;
+          }
+        }
+      }
+
+      if (count > 0) {
+        deleteBatch(mapImpl, schema, AggregatorUtil.ingestAggregate(batch, schema));
+      }
+    }
+  }
+
+  private void deleteBatch(final MapImpl mapImpl, final Schema schema, final Iterable<? extends Element> elements) {
+    for (final Element element : elements) {
+      if (null != element) {
+        final Element elementForIndexing = deleteElement(element, schema, mapImpl);
+
+        // Update entityIdToElements and edgeIdToElements if index required
+        if (mapImpl.isMaintainIndex()) {
+          updateElementIndex(elementForIndexing, mapImpl);
+        }
+      }
+    }
+  }
+
+  private Element deleteElement(final Element element, final Schema schema, final MapImpl mapImpl) {
```

Comment:
Not sure if this is intended but does this handle removing edges with dead ends? Say if you delete a node with an edge on does that edge also get deleted as technically one end of the edge is now invalid?

Reply:
Not currently. We spoke to @GCHQDeveloper314 I think the intention for this MVP was to just delete exactly the input and then we can improve on that if required. It will be up to the user to make sure they are deleting the correct elements by just running GetElements first to check everything they want to delete is there and then adding the DeleteElements to the operation chain.